frontend/packages/dev-console/src/components/pipelineruns/PipelineRunLogs.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import { Nav, NavItem, NavList } from '@patternfly/react-core';
+import { PodLogs } from '@console/internal/components/pod-logs';
+import { StatusIcon } from '@console/shared';
+import { Firehose } from '@console/internal/components/utils';
+import { pipelineRunFilterReducer } from '../../utils/pipeline-filter-reducer';
+import { PipelineRun } from '../../utils/pipeline-augment';
+import './PipelineRunLogs.scss';
+
+interface PipelineRunLogsProps {
+  obj: PipelineRun;
+}
+interface PipelineRunLogsStates {
+  activeItem: string;
+}
+class PipelineRunLogs extends React.Component<PipelineRunLogsProps, PipelineRunLogsStates> {
+  constructor(props) {
+    super(props);
+    this.state = { activeItem: null };
+  }
+
+  componentDidMount() {
+    const { obj } = this.props;
+    const taskRunFromYaml =
+      obj.status && obj.status && obj.status.taskRuns ? obj.status.taskRuns : {};
+    const taskRuns = this.getSortedTaskRun(taskRunFromYaml);
+    this.setState({ activeItem: taskRuns[taskRuns.length - 1] });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props !== nextProps) {
```

Comment:
Better to compare `obj` here and make it more specific. Make sure that shallow compare is good enough as the condition in your case.